c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -3123,12 +3123,30 @@ class DeviceCachingAllocator {
 // Returns whether to force all allocations to bypass the caching allocator and
 // go straight to cudaMalloc.  This setting is useful when debugging GPU memory
 // errors, since the caching allocator foils cuda-memcheck.
-bool forceUncachedAllocator() {
-  static bool force_uncached =
-      getenv("PYTORCH_NO_CUDA_MEMORY_CACHING") != nullptr;
+static bool forceUncachedAllocator() {
+  // Allow either CUDA or HIP name for env var for maximum user comfort
+  // break up the name to avoid hipify changing it
+  static const char* cuda_env = getenv(
+      "PYTORCH_NO_C"
+      "UDA_MEMORY_CACHING");
```

Comment:
nit: can we not break into two lines?

Reply:
I wanted to but lintrunner complained.  I'll instead protect it from getting hipified by putty a dummy mapping into the cuda-to-hip mappings file.